utils/python/CIME/XML/generic_xml.py

```diff
@@ -57,21 +63,27 @@ def get_node(self, nodename, attributes=None, root=None):
         """
         if(root is None):
             root = self.root
-
+        nodes = []
         xpath = ".//"+nodename
         if(attributes is not None):
             keys = list(attributes.keys())
-            cnt = 0
+            # xml.etree has limited support for xpath and does not allow more than
```

Comment:
Sorry about the xpath stuff. Python's built-in library support is usually very comprehensive, so it's odd that the xpath support is limited.